src/font-stylesheet-timeout.js

```diff
@@ -49,20 +57,33 @@ export function fontStylesheetTimeout(win) {
 
   // Avoid timer dependency since this runs very early in execution.
   win.setTimeout(() => {
-    // We waited for the timeout period. There is no way to check whether
-    // the stylesheet actually loaded. For that reason we check whether
-    // the document is ready instead. The link tags block the readiness
-    // and they are the only external resource that does, so if the doc
-    // isn't ready yet it is probably the stylesheet's fault.
-    if (isDocumentReady(win.document)) {
+    const styleSheets = win.document.styleSheets;
+    if (!styleSheets) {
       return;
     }
-    // Alright we timed out.
-    // Find all stylesheets.
+    // Find all stylesheets that aren't loaded from the AMP CDN (those are
+    // critical if they are present).
     const styleLinkElements = win.document.querySelectorAll(
-        'link[rel~="stylesheet"]');
+        'link[rel~="stylesheet"]:not([href^="' + urls.cdn + '"])');
+    // Compare external sheets against elements of document.styleSheets.
+    // They do not appear in this list until they have been loaded.
+    const timedoutStyleSheets = [];
     for (let i = 0; i < styleLinkElements.length; i++) {
-      const existingLink = styleLinkElements[i];
+      const link = styleLinkElements[i];
+      let found = false;
+      for (let n = 0; n < styleSheets.length; n++) {
```

Comment:
Nit: `styleSheets` is a host object, and iterating it might be slow. We could do a `toArray` on it to begin with.